src/test/java/uk/gov/companieshouse/api/accounts/controller/PreviousPeriodControllerTest.java

```diff
@@ -164,7 +163,62 @@ public void canRetrievePreviousPeriod() throws DataException {
 
         assertNotNull(response);
         assertEquals(HttpStatus.OK, response.getStatusCode());
-        assertEquals(previousPeriod, response.getBody());
+    }
+
+    @Test
+    @DisplayName("PUT - Tests the successful update of a previous period resource")
+    public void canUpdatePreviousPeriod() throws DataException {
+        mockSmallFull();
+        mockPreviousPeriodLinkOnSmallFullResource();
+
+        when(previousPeriodValidator.validatePreviousPeriod(any())).thenReturn(errors);
+        ResponseObject responseObject = new ResponseObject(ResponseStatus.UPDATED, previousPeriod);
+        ResponseEntity responseEntity = ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+        when(apiResponseMapper.map(responseObject.getStatus(),null, responseObject.getValidationErrorData()))
+                .thenReturn(responseEntity);
+        doReturn(responseObject).when(previousPeriodService).update(previousPeriod, null, "12345", null);
+
+        ResponseEntity response = previousPeriodController.update(previousPeriod, bindingResult, "12345", request);
+
+        assertNotNull(response);
+        assertEquals(HttpStatus.NO_CONTENT, response.getStatusCode());
+    }
+
+    @Test
+    @DisplayName("PUT - Tests the unsuccessful update of a previous period resource due to no link to small full resource")
+    public void canUpdatePreviousPeriodFail() {
+        mockSmallFull();
+        when(smallFull.getLinks()).thenReturn(new HashMap<>());
+        ResponseEntity response = previousPeriodController.update(previousPeriod, bindingResult, "123456", request);
```

Comment:
we could use a constant for this `123456` instead of literal string.

Reply:
Sorry, I merged this before I saw this comment. But yes we could have done it as a literal string to reduce duplicate :+1: my bad